app/jobs/application_job.rb

```diff
@@ -2,4 +2,18 @@ class ApplicationJob < ActiveJob::Base
   def humanized_name
     self.class.name
   end
+
+  def self.spawn_if_missing(world)
+    return if (Foreman.in_rake? && !Foreman.in_rake?('dynflow:executor')) || Rails.env.test?
+
+    pending_jobs = world.persistence.find_execution_plans(filters: { :state => 'scheduled' })
```

Comment:
The different shape of args should be handled [here](https://github.com/theforeman/foreman/pull/5575/files#diff-ca33478ffbe0ef63784b3a32a155a83eR13), this is actually an already fixed version because we encountered things breaking on that line exactly for the reason you mentioned.

Tests are always a good idea, I'll see what I can do about it

Reply:
That's different case. The problem is, when you do delayed task outside of active job, where the `:serialized_args` could be something like `[1,2,3]`, and `delayed_plan.to_hash[:serialized_args].first.try(:[], 'job_class')` would fail on `1` not defining method `[]`

Reply:
Ah, good point